### PR TITLE
Do not abort on chassis query errors

### PR DESCRIPTION
**Pull request by kolsa**
Prevents also panicking spotted in [this](https://github.com/metal-stack/metal-core/issues/24) metal-core issue.

### CODE REVIEW

File: internal/redfish/redfish.go [hunk 1]
```diff
@@ -51,6 +52,9 @@ func New(url, user, password string, insecure bool) (*APIClient, error) {
 
 func (c *APIClient) BoardInfo() (*api.Board, error) {
 	// Query the chassis data using the session token
+	if c.Service == nil {
+		return nil, errors.New("gofish service root is not available most likely due to missing username")
```
Comment (mwindower) [inline internal/redfish/redfish.go:56]: is there any chance to set the missing username?

Comment (kolsa) [reply internal/redfish/redfish.go:56]: go-hal could use the superuser credentials if the given username is empty.
We can do so as soon as [this](https://github.com/metal-stack/go-hal/pull/7) PR is merged.

**Review by mwindower — APPROVED** [review]
lgtm

**Comment by majst01** [comment]
We should take this in our  current shared networks branch of metal-core !

**Pull request merged**